File: codeforces/1600~1699/1681/c.cc
#include <bits/stdc++.h>

using namespace std;

using i64 = long long;

void solve() {
  int n = 0;
  cin >> n;
  vector<int> a(n);
  for (int i = 0; i < n; ++i) cin >> a[i];
  vector<int> b(n);
  for (int i = 0; i < n; ++i) cin >> b[i];

  vector<pair<int,int>> ans;
  vector<int> c = a;
  sort(c.begin(), c.end());
  for (int i = 0; i < n; ++i) {
    if (a[i] != c[i]) {
      for (int j = i+1; j < n; ++j) {
        if (a[j] == c[i]) {
          swap(a[i], a[j]);
          swap(b[i], b[j]);
          ans.emplace_back(i+1, j+1);
          break;
        }
      }
    }
  }

  c = b;
  sort(c.begin(), c.end());
  for (int i = 0; i < n; ++i) {
    if (b[i] != c[i]) {
      for (int j = i+1; j < n; ++j) {
        if (b[j] == c[i]) {
          swap(a[i], a[j]);
          swap(b[i], b[j]);
          ans.emplace_back(i+1, j+1);
          break;
        }
      }
    }
  }

  for (int i = 1; i < n; ++i) {
    if (a[i-1] > a[i]) {
      cout << "-1" << endl;
      return;
    }
  }

  cout << ans.size() << endl;
  for (auto& [i, j] : ans) {
    cout << i << " " << j << endl;
  }
}

int main(void) {
  ios::sync_with_stdio(false);
  cin.tie(0);

  int T = 0;
  cin >> T;
  for (int t = 0; t < T; ++t) {
    solve();
  }
  return 0;
}

File: codeforces/1600~1699/1681/c.rs
use std::io;
use std::str::FromStr;
use std::fmt::Debug;

fn read_int<T: FromStr>() -> T where <T as FromStr>::Err: Debug {
    let mut s = String::new();
    io::stdin().read_line(&mut s).unwrap();
    s.trim().parse::<T>().expect("invalid input")
}

fn read_vec<T: FromStr>() -> Vec<T> where <T as FromStr>::Err: Debug {
    let mut s = String::new();
    io::stdin().read_line(&mut s).unwrap();
    s.split(" ").map(|x| x.trim().parse().expect("invalid input")).collect()
}

fn solve() {
    let n = read_int::<usize>();
    let mut a = read_vec::<i32>();
    let mut b = read_vec::<i32>();

    let mut ans: Vec<(usize,usize)> = Vec::new();

    let mut c = a.clone();
    c.sort();
    for i in 0..n {
        if a[i] != c[i] {
            for j in (i+1)..n {
                if a[j] == c[i] {
                    a.swap(i, j);
                    b.swap(i, j);
                    ans.push((i+1, j+1));
                    break;
                }
            }
        }
    }

    c = b.clone();
    c.sort();
    for i in 0..n {
        if b[i] != c[i] {
            for j in (i+1)..n {
                if b[j] == c[i] {
                    a.swap(i, j);
                    b.swap(i, j);
                    ans.push((i+1, j+1));
                    break;
                }
            }
        }
    }

    for i in 1..n {
        if a[i-1] > a[i] {
            println!("-1");
            return;
        }
    }

    println!("{}", ans.len());
    for i in 0..ans.len() {
        println!("{} {}", ans[i].0, ans[i].1);
    }
}

fn main() {
    let n = read_int::<i32>();
    for _t in 0..n {
        solve();
    }
}
